tests/virtual_machines_test.go

```diff
@@ -450,6 +449,63 @@ var _ = Describe("Virtual Machines", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
+
+		Context("When a VM's NIC is removed and a new VM is created with the same MAC", func() {
+			It("should successfully release the MAC and the new Vm should be created with no errors", func() {
+				err := setRange(rangeStart, rangeEnd)
+				Expect(err).ToNot(HaveOccurred())
+
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", ""), newInterface("br2", "")},
+					[]kubevirtv1.Network{newNetwork("br1"), newNetwork("br2")})
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), vm)
+
+				}, 50*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+				mac1, err := net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+				mac2, err := net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[1].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+
+				updatedVM := *vm.DeepCopy()
+				By("before the NIC is deleted")
+				newVm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", mac1.String())},
+					[]kubevirtv1.Network{newNetwork("br1")})
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), newVm)
+
+				}, 50*time.Second, 5*time.Second).Should(HaveOccurred(), "should have failed to create the vm because the mac is allready taken")
+
+				By("deleting the VM's NIC")
+				Eventually(func() error {
+					err := testClient.VirtClient.Get(context.TODO(),
+						client.ObjectKey{Namespace: updatedVM.Namespace, Name: updatedVM.Name}, &updatedVM)
+					if err != nil {
+						return err
+					}
+					updatedVM.Spec.Template.Spec.Domain.Devices.Interfaces = []kubevirtv1.Interface{newInterface("br2", mac2.String())}
+					updatedVM.Spec.Template.Spec.Networks = []kubevirtv1.Network{newNetwork("br2")}
+					err = testClient.VirtClient.Update(context.TODO(), &updatedVM)
+					if err != nil {
+						return err
+					}
+
+					if len(updatedVM.Spec.Template.Spec.Domain.Devices.Interfaces) != 1 {
+						return fmt.Errorf("failed to delete the VM's interface")
+					}
+					return nil
+
+				}, 50*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to update client")
+
+				By("after the VM's NIC has been deleted")
+				Expect(updatedVM.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).To(Equal(mac2.String()),
+					"kubemacppool should assign the same mac as before the update")
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), newVm)
```

Comment:
here you need to create a new vm not the one you already use.

Reply:
but I think the creation of "newVm" should fail, so it is not really used, isn't it?

Reply:
oh you are right miss that we have vm, newVm and updateVM haha